metric_learn/_util.py

```diff
@@ -405,3 +405,15 @@ def validate_vector(u, dtype=None):
   if u.ndim > 1:
     raise ValueError("Input vector should be 1-D.")
   return u
+
+
+def _check_num_dims(n_features, num_dims):
+  """Checks that num_dims is less that n_features and deal with the None
+  case"""
+  if num_dims is None:
+    dim = n_features
+  else:
+    if not 0 < num_dims <= n_features:
+      raise ValueError('Invalid num_dims, must be in [1, %d]' % n_features)

```

Comment:
Some existing code would only warn if `num_dims > n_features`. I think it's probably better to error out here, but we should keep in mind that this is technically a back-compat break.

Reply:
That's true, maybe adding it in the changelog is enough ? Something like "for all the algorithms that have a parameter `num_dims`, it will now be checked to be between 1 and `n_features`, with `n_features` the number of dimensions of the input space" ?

Reply:
I just added it to the ![release draft](https://github.com/metric-learn/metric-learn/releases/tag/untagged-6a42a4c7c78fcdc3daec)